src/modules/new-request-form/fields/attachments/Attachments.tsx

```diff
@@ -0,0 +1,192 @@
+import {
+  FileUpload,
+  Field as GardenField,
+  Hint,
+  Input,
+  Label,
+  Message,
+} from "@zendeskgarden/react-forms";
+import {
+  Close,
+  Notification,
+  Title,
+  useToast,
+} from "@zendeskgarden/react-notifications";
+import { useCallback } from "react";
+import { useDropzone } from "react-dropzone";
+import type { Field } from "../../data-types";
+import type { AttachedFile } from "./AttachedFile";
+import { FileListItem } from "./FileListItem";
+import { useAttachedFiles } from "./useAttachedFiles";
+
+interface AttachmentProps {
+  field: Field;
+}
+
+async function fetchCsrfToken() {
+  const response = await fetch("/api/v2/users/me.json");
+  const {
+    user: { authenticity_token },
+  } = await response.json();
+  return authenticity_token as string;
+}
+
+export interface UploadFileResponse {
+  upload: {
+    attachment: {
+      file_name: string;
+      content_url: string;
+    };
+    token: string;
+  };
+}
+
+export function Attachments({ field }: AttachmentProps): JSX.Element {
+  const { label, error, name, attachments, description } = field;
+  const {
+    files,
+    addPendingFile,
+    setPendingFileProgress,
+    setUploaded,
+    removePendingFile,
+    removeUploadedFile,
+  } = useAttachedFiles(
+    attachments?.map((value) => ({
+      status: "uploaded",
+      value,
+    })) ?? []
+  );
+
+  const { addToast } = useToast();
+
+  const notifyError = useCallback(
```

Comment:
Do we need the `useCallback` in this case?

Reply:
It seems that the `onDrop` from `react-dropzone` needs the `useCallback`: https://react-dropzone.js.org/. Our `onDrop` is calling `notifyError` so it needs to be passed in the dependency array of it and memorized as well.